core/integration/module_go_test.go

```diff
@@ -430,6 +430,57 @@ func (m *HasNotMainGo) Hello() string { return "Hello, world!" }
 		require.NoError(t, err)
 		require.Contains(t, sourceSubdirEnts, "go.mod", "go.sum")
 	})
+
+	t.Run("init module in .dagger if files present in current dir", func(ctx context.Context, t *testctx.T) {
```

Comment:
Ah sorry, should have caught this before - I think we should add a test for a `init`+`develop`.

How does it handle:

```
$ dagger init --name=foo
$ ls -a
.  ..  dagger.json
$ dagger develop --sdk=go
```

`dagger.json` should be ignored, and we should put it in `.` instead of `.dagger`.